server/src/main/java/mineclone/common/world/block/BasicWireBlock.java

```diff
@@ -0,0 +1,275 @@
+		for (Direction dir : Direction.HORIZONTAL)
+			state = updateConnection(world, pos, state, dir);
+
+		return resolveState(state);
+	}
+
+	private IBlockState updateConnection(IWorld world, IBlockPosition pos, IBlockState state, Direction dir) {
+		WireSide connection = findConnection(world, pos, dir);
+		return state.with(CONNECTIONS.get(dir), connection);
+	}
+
+	private WireSide findConnection(IWorld world, IBlockPosition pos, Direction dir) {
+		ConnectionSide side = ConnectionSide.fromDirection(dir);
+
+		if (type.getConnection(world, pos, side.withUp()) != ConnectionType.NONE) {
+			Direction opp = dir.getOpposite();
+			IBlockPosition adjacent = pos.offset(dir);
+			IBlockState adjacentState = world.getBlockState(adjacent);
+
+			return adjacentState.isAligned(opp) ? WireSide.UP : WireSide.SIDE;
+		}
+		if (type.getConnection(world, pos, side) != ConnectionType.NONE) {
+			return WireSide.SIDE;
+		}
+		if (type.getConnection(world, pos, side.withDown()) != ConnectionType.NONE) {
+			return WireSide.SIDE;
+		}
+
+		return WireSide.NONE;
+	}
+
+	private IBlockState resolveState(IBlockState state) {
+		Direction connectionDir = null;
+
+		for (Direction dir : Direction.HORIZONTAL) {
+			WireSide connection = state.get(CONNECTIONS.get(dir));
+
+			if (connection != WireSide.NONE) {
+				if (connectionDir != null) {
+					// There is only something to resolve, if there is
+					// exactly zero or one connection in the state.
+					return state;
+				}
+
+				connectionDir = dir;
+			}
+		}
+
+		if (connectionDir == null) {
+			for (Direction dir : Direction.HORIZONTAL)
+				state = state.with(CONNECTIONS.get(dir), WireSide.SIDE);
+		} else {
+			Direction dir = connectionDir.getOpposite();
+
+			state = state.with(CONNECTIONS.get(dir), WireSide.SIDE);
+		}
+
+		return state;
+	}
+
+	@Override
+	public WireType getWireType() {
+		return type;
+	}
+
+	@Override
+	public int getSignal(IServerWorld world, IBlockPosition pos, IBlockState state, Direction dir) {
+		if (dir == Direction.UP) {
+			return type.min();
+		} else if (dir.isHorizontal()) {
+			WireSide connection = state.get(CONNECTIONS.get(dir));
+
+			if (connection == WireSide.NONE) {
+				return type.min();
+			}
+		}
+
+		return getSignal(state);
+	}
+
+	@Override
+	public int getDirectSignal(IServerWorld world, IBlockPosition pos, IBlockState state, Direction dir) {
+		return getSignal(world, pos, state, dir);
+	}
+
+	@Override
+	public int getSignal(IBlockState state) {
+		return state.get(POWER);
+	}
+
+	@Override
+	public IBlockState setSignal(IBlockState state, int signal) {
+		return state.with(POWER, signal);
+	}
+
+	@Override
+	protected IBlockState createDefaultState() {
+		return BlockState.createStateTree(this, POWER,
+				                                NORTH_CONNECTION,
+				                                SOUTH_CONNECTION,
+				                                WEST_CONNECTION,
+				                                EAST_CONNECTION);
```

Comment:
Incorrect indentation spacing. I suggest the following:

```suggestion
		return BlockState.createStateTree(this, POWER,
		                                        NORTH_CONNECTION,
		                                        SOUTH_CONNECTION,
		                                        WEST_CONNECTION,
		                                        EAST_CONNECTION);
```